### ISSUE TITLE
Support

### ISSUE THREAD

**Issue by brucou** [radian-software/riju-replit#1]
Hi there,

I am sorry to hear about the legal disputes with repl.it That certainly does not inspire to use their products even the free tier. In the end, company behavior is part of the product too.

I can't tell you what to do there, it is scary to be legally threatened. That is the whole point. That said, the fact that the threat is issued does not mean that the threat has merits, and that's what layers are for. I am unaware of NGO that may help with free legal advice when it comes to IP and plagiarism but there should be? I am sorry I can't help but I think it is worth checking your legal options if that project goes deep to your heart. It is a legal dispute, it is not a moral thing. If you think you are right, investigate the availability of free legal advice. If Maximiliam Schrems did not take on Facebook, we would have made less progress on privacy and data protection.

Anyways it is easy to write this and have opinions. It is another thing to be in the concrete situation. I just wanted to commnicate that that the merits of a case have nothing to do with the energy with which you are threatened.  If they open a case against you they will have to make their case, and the case that is there according to the emails you shown looks unclear at best. The legal bar for plagiarism is actually not that low. I would at least have them produce their (legal) argumentation. But of course I am not a laywer...

**Comment by kexik** [comment]
If you haven't stolen any of their code directly or some internal, non-public ideas you should not worry! I am sure a lot of people would support you. You could even start a crowdfunding for your legal expenses or ask EFF/FSF for help.

The fact that a startup has millions of funding doesn't give it power to block any project similar to theirs. If that were true, there wouldn't be Linux, Macos and others because they all have concept of windows.

**Comment by gaetgu** [comment]
Saw this on hacker news - I wish it was still up! 

I have really loved repl.it over the years, but this is just wrong. For a company that claims to support open source, it is not right that they should shut down someone's non-competitive hobby. 

I hope that you can get riju online sometime soon, I will definitely use it!

**Comment by hirako2000** [comment]
Oh well.. [github.com/hirako2000/riju](https://github.com/hirako2000/riju)

**Comment by twerq** [comment]
Yeah, I've cloned fork by [umesh-timalsina/riju](https://github.com/umesh-timalsina/riju) to my local drive. For the sake of possibility =)

**Comment by twerq** [comment]
Yeah, I've cloned fork [umesh-timalsina/riju](https://github.com/umesh-timalsina/riju) to my local drive. For the sake of possibility =)

**Comment by aeiouaeiouaeiouaeiouaeiouaeiou** [comment]
[Gitee mirror](https://gitee.com/aeiouaeiouaeiouaeiouaeiouaeiou/riju)

**Comment by EwanValentine** [comment]
Just a quick message of support after reading your blog post: I won't be using Repl.it again soon. Their response to this project was unfair and classless. Running code in a browser is not intellectual property, it's no big secret, as you pointed out, there are literally hundreds of similar platforms already out there, that preceded Repl.it. If your entire market position comes down to the n of X you have, then that's an intrinsically weak position in terms of IP, and that's the only part of Repl.it I can see are similar. As you pointed out, Repl.it's value prop is all the 'other stuff'. Pains me to see companies using their success to try to crush individuals. I'm sure they benefit from lots of open-source work themselves. Even if there was a strong copyright claim, they could have handled it better. I'm glad you received an apology, and hope their actions back it up now.

**Comment by gaetgu** [comment]
Once you get this back up I will be transitioning to only this website instead of repl.it (and maybe clone it to add workspace persistence on a self-hosted server or something).

**Comment by rayhanadev** [comment]
Wow, being a Replit power-user and fanboy I am very distressed to hear about this darker side of Replit...

**Comment by hirako2000** [comment]
I'm also doing a 180 degree turn on my view on repl.it.
Fyi I'm looking at standing up a full or near support of all these languages as a free instance (based on this repo).

estimate costs would be about $20 a month. with low traffic. If someone want to collaborate with this, provide support, it be to fund it or tech assistance, dm me.

**Comment by rayhanadev** [comment]
> I'm also doing a 180 degree turn on my view on repl.it.
> Fyi I'm looking at standing up a full or near support of all these languages as a free instance (based on this repo).
> 
> estimate costs would be about $20 a month. with low traffic. If someone want to collaborate with this, provide support, it be to fund it or tech assistance, dm me.

@hirako2000

Replit now has Nix so practically all languages are supported, and it is all free getting the Hacker plan only improves speeds, storage, and gives you a few other extraneous features. I'd be willing to help out frontend wise if I'm familiar with your tech stack lol.

**Comment by hirako2000** [comment]
@RayhanADev, thanks for pointing that out.

Unbiased opinions on the relevance of making open source service like ruju available again, or ruju itself at scale, as an alternative to repl.it would be useful though :)

**Comment by mohkale** [comment]
Hi @raxod502 

I don't mean to bother you but the last update to your [blog post](https://intuitiveexplanations.com/tech/replit/) mentioned you were going to put riju back up. Is that still happenning?

**Comment by raxod502** [comment]
Yes, that's absolutely still happening. I'm still working on some infrastructure upgrades that will be needed to handle the increased traffic. And I'm also on vacation in Hawaii. But rest assured I do plan to get things back up. I just want everything to be fully working and solid before I share.

**Comment by raxod502** [comment]
Tada:

* https://github.com/raxod502/riju
* https://riju.codes/

The old fork that's been floating around was ~7,000 lines of code versus the ~18,500 in this new version, so I hope you'll forgive the delay :)

Please feel free to open issues for any bugs you find. I made some infrastructural upgrades (autoscaling, metrics & alerting, etc.), so hopefully I can keep it from going down immediately with nontrivial traffic, but at the same time I kind of expect it to go down immediately nonetheless...
